source/MaterialXRenderGlsl/TextureBaker.cpp

```diff
@@ -482,67 +480,73 @@ BakedDocumentVec TextureBaker::createBakeDocuments(DocumentPtr doc, const FileSe
         genContext.registerSourceCodeSearchPath(path / "libraries");
     }
     genContext.getShaderGenerator().setColorManagementSystem(cms);
-    StringResolverPtr resolver = StringResolver::create();
-    StringVec renderablePaths = getRenderablePaths(doc);
 
-    BakedDocumentVec bakedDocuments;
-    for (const string& renderablePath : renderablePaths)
+    // Compute the material tag set.
+    StringVec materialTags = udimSet;
+    if (materialTags.empty())
+    {
+        materialTags.push_back(EMPTY_STRING);
+    }
+
+    if (_outputStream)
+    {
+        *_outputStream << std::endl << "Working on material: " << materialPath << std::endl;
+    }
+
+    NodePtr shaderNode;
```

Comment:
What are your thoughts on this `shaderNode` variable, which is declared outside of the material tag loop, but defined multiple times within it?  If it depends only upon the `materialPath` that is passed into the method, then perhaps it could be computed entirely outside of the loop, thus simplifying this logic?